app/controllers/catalog_controller.rb

```diff
@@ -5,6 +5,30 @@ class CatalogController < ApplicationController
 
   include Blacklight::Catalog
 
+  # Destroy a layer if user is admin and layer is found
+  def destroy
+    if current_user and current_user.admin
+      solr = Blacklight.default_index.connection
+      response = solr.get 'get', :params => { :id => params[:id] }
+      if (response["doc"] == nil)
+        flash[:error] = 'Sorry, the layer you are trying to delete does not exist.'
+        redirect_to root_path
+      else
+        response = solr.delete_by_query "layer_slug_s:#{params[:id]}"
+        solr.commit
+        if (response["responseHeader"]["status"] == 0)
+          flash[:success] = 'Layer deleted from solr.'
+          redirect_to root_path
+        else
+          flash[:error] = 'Sorry. Not able to delete layer from solr.'
```

Comment:
Is it possible to log the error somewhere if it fails? We don't need to show the user if but maybe if RAILS_ENV=development we could log this to the `application.log` or something...not sure how to do that though

Reply:
I feel it is necessary to show the user that there was some internal error in the process. It would be confusing for the user when they'd see the same layer again which was supposedly deleted.

Reply:
oh I mean we don't need to show the user the full text of the error, sorry. I think we should:
- log the error (maybe a trace?) to some log
- show the flash message, which you do
- redirect to the layer page.